Cosmic's Golf Course

cosmiccoincidence opened the pull request:
Adds..
-Golfcourse (map + overmap special + overmap terrain)
-golf gloves (item)
-golf hat (item)
-golf bag (item)
-golf shoes (item)
-golf tee (item)
-golf ball (item)
-above items to various item groups
-golf cart (item group)
-golf green (item group)
-long grass (terrain)
-dead grass (terrain)
-golf green (terrain)
-golf hole (terrain)
-golf cart 4 seat varient (vehicle)
-golf carts (vehicle group)

Xhuis commented:
Might not be a bad idea to add some of the clothes to clothing stores and sporting goods stores. Just s suggestion, though.

John-Candlebury commented:
You should make the golfbag be a scabbard for golfing irons!

Fore!

Night-Pryanik commented on data/json/item_groups.json:
```diff
@@ -208,6 +208,12 @@
       ["fan", 10],
       ["pipe_glass", 5],
       ["coke", 8],
+	  [ "hat_golf", 15 ],
```
Indent here and below.

Night-Pryanik commented on data/json/items/armor.json:
```diff
@@ -9666,5 +9666,64 @@
+    "id": "golf_bag",
+    "type": "ARMOR",
+    "name": "golf_bag",
+    "description": "A tall canvas and plastic bag with fold-out legs used for golfing. It even has straps to be worn on the back.",
```
Two spaces after periods.

Night-Pryanik commented on data/json/items/armor/boots.json:
```diff
@@ -501,6 +501,26 @@
+    "type": "ARMOR",
+    "name": "pair of golf shoes",
+    "name_plural": "pairs of golf shoes",
+    "description": "Snug fit sport shoes with hard ridges on the bottom",
```
Period at the end.

Night-Pryanik commented on data/json/overmap_specials.json:
```diff
@@ -2240,5 +2240,29 @@
+      { "point": [ 0, 1, 0 ], "overmap": "golfcourse_01_north" },
+      { "point": [ 0, 2, 0 ], "overmap": "golfcourse_02_north" },
+      { "point": [ 1, 0, 0 ], "overmap": "golfcourse_10_north" },
+	  { "point": [ 1, 1, 0 ], "overmap": "golfcourse_11_north" },
```
Indent here and below.

Night-Pryanik commented on data/json/vehicles/carts.json:
```diff
@@ -97,9 +97,48 @@
+      { "x": 0, "y": 0, "part": "seat" },
+      { "x": 0, "y": 0, "part": "roof" },
+      { "x": 0, "y": 0, "part": "engine_electric" },
+      { "x": 0, "y": 0, "part": "controls" },
```
Add a dashboard, or one wouldn't be able to control electrics of the car.

cosmiccoincidence replied:
I tested both the original and my new golf cart and both operate just fine. Also I made the other changes.

cosmiccoincidence commented:
- No need to add items to clothing and sporting stores as I added them to groups such as "allclothes", "hatstore_hats", "shoestore_shoes", "shoes", "livingroom", "sports", and "allsporting" which is what those locations use.

- Made golf bag a sheath for golf clubs.

-Fixed Pryanik's suggestions.

cosmiccoincidence commented:
Had to tweak that golf bag sheath a bit. But its finished now.

cosmiccoincidence commented on data/json/construction.json:
```diff
@@ -2096,13 +2096,58 @@
+        "category" : "OTHER",
+        "required_skills" : [ [ "survival", 0 ] ],
+        "time" : 6,
+        "qualities": [ [ { "id": "CUT", "level": 1 } ] ],
```
This was actually how it was before I made my addition (check out your local version of construction.json. But I am willing to change it for the better.

cainiaowu replied:
Or add a GRASSCUTTING quality and add it to those tools.

cosmiccoincidence replied:
@cainiaowu I'm not sure where I add new qualities, if I could get pointed in the right direction I'll look into it.

cosmiccoincidence commented:
@Night-Pryanik 
Let me know what you think of the fix.

Night-Pryanik commented on data/json/construction.json:
```diff
@@ -2136,7 +2138,7 @@
         "required_skills" : [ [ "survival", 0 ] ],
         "time" : 6,
-        "qualities": [ [ { "id": "CUT", "level": 1 }, { "id": "DIG", "level": 1 } ] ],
+        "qualities": [ [ { { "id": "DIG", "level": 1 } ] ],
```
Excessive `{`.

Night-Pryanik commented on data/json/construction.json:
```diff
@@ -2145,7 +2147,7 @@
         "required_skills" : [ [ "survival", 0 ] ],
         "time" : 6,
-        "qualities": [ [ { "id": "CUT", "level": 1 }, { "id": "DIG", "level": 1 } ] ],
+        "qualities": [ [ { { "id": "DIG", "level": 1 } ] ],
```
The same here.

Night-Pryanik commented on data/json/construction.json:
```diff
@@ -2100,16 +2100,18 @@
-        "post_terrain" : "t_grass"
+        "post_terrain" : "t_grass",
+        "tools": [
+        [ [ "scythe", 2 ], [ "sickle", 2 ] ]
```
Remove `2`s, as it would try to consume 2 charges of scythe or sickle, and it's good to go.
